contracts/finance/VestingWallet.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../token/ERC20/extensions/ERC20Votes.sol";
+import "../token/ERC20/utils/SafeERC20.sol";
+import "../utils/Context.sol";
+
+/**
+ * @title VestingWallet
+ * @dev This contract handles the vesting of ERC20 tokens for a given beneficiary. Custody of multiple tokens can be
+ * given to this contract, which will release the token to the beneficiary following a given vesting schedule. The
+ * vesting scheduled is customizable through the `vestedAmount(address,uint256)` function.
+ *
+ * Any token transferred to this contract will follow the vesting schedule as if they were locked from the beginning.
+ * Consequently, if the vesting has already started, any amount of tokens sent to this contract will (at least partly)
+ * be immediately releasable.
+ *
+ * While tokens are locked, the beneficiary still has the ability to delegate the voting power potentially associated
+ * with these tokens.
+ */
+contract VestingWallet is Context {
+    event TokensReleased(address token, uint256 amount);
+
+    mapping(address => uint256) private _released;
+    address private immutable _beneficiary;
+    uint256 private immutable _start;
+    uint256 private immutable _duration;
+
+    modifier onlyBeneficiary() {
+        require(beneficiary() == _msgSender(), "VestingWallet: access restricted to beneficiary");
+        _;
+    }
+
+    /**
+     * @dev Set the beneficiary, start timestamp and vesting duration of the vesting wallet.
+     */
+    constructor(
+        address initialBeneficiary,
+        uint256 initialStart,
+        uint256 initialDuration
```

Comment:
Sorry I don't think these are good because it can mislead into thinking the values could change later. I would either keep the underscore suffixes or go with `beneficiaryAddress`, `startTimestamp`, `durationSeconds`.